### ISSUE TITLE
Failed to write bootloader configuration

### ISSUE THREAD

**Issue by Tjzabel** [RITlug/TigerOS#58]
@axk4545  @ct-martin When installing TigerOS GNOME edition to my physical laptop, the bootloader fails to install properly.

When I select the exact same configuration on the official Fedora 26 distro, the installation goes fine.

**Comment by ct-martin** [comment]
Debugged w/ @Tjzabel - Confirmed issue exists and involves UEFI, BIOS install works fine.

@axk4545 going to assign this to you since this is your territory.

**Comment by axk4545** [comment]
@ct-martin Will look into it soo.

**Comment by Tjzabel** [comment]
UPDATE: Currently this issue is present both in the Cinnamon and GNOME TigerOS installations.

**Comment by linux-modder** [comment]
@axk4545 @Tjzabel confirm installer / install image have grub2-efi package installed.  If so, file bug upstream with both anaconda and grub2 developers.

**Comment by axk4545** [comment]
@linux-modder @Tjzabel @ct-martin it is in the ks.  please confirm it is actually in the resultant image.

**Comment by axk4545** [comment]
per @linux-modder add --efi to LMC command in build scripts.

**Comment by ct-martin** [comment]
@axk4545 Can confirm grub2-efi is installed on the liveusb, will rebuild with —efi and then we can retest

**Comment by ct-martin** [comment]
"livemedia-creator: error: unrecognized arguments: --efi"
(cc @axk4545 @linux-modder )

**Comment by axk4545** [comment]
This makes me think we should also test on Mac as they have a special EFI implementation

**Comment by axk4545** [comment]
@axk4545 Attempted an install with a TigerOS live USB burned with Fedora Media Writer. Same result

**Comment by linux-modder** [comment]
@ct-martin on live doesn't mean it's in the squashfs tho the squahsfs is what is rsync'd off for install on a non netinstall iso

Will check the present ks today and see if I can find a bug in the logic or usage.

**Comment by ct-martin** [comment]
@linux-modder do you have a command I can run on the live to test if it's there?

**Comment by axk4545** [comment]
@Tjzabel will test once we get a build. currently blocked on gstreamer dep issues upstream

**Comment by axk4545** [comment]
@Tjzabel can I have the logs from the install and the exact error message? the logs will be at `/tmp/anaconda.log`, `/tmp/program.log` and `/tmp/syslog` on the installer system(when running from the install disk/live usb). they can also be pulled from the installed system in `/var/log/anaconda` if you skip the GRUB step and complete the install.

**Comment by axk4545** [comment]
It might be an issue with the anaconda install class. I will look at it.

**Comment by Tjzabel** [comment]
Soooo The UEFI build works!! How should I proceed? This issue may be available to close.

**Comment by axk4545** [comment]
@Tjzabel test it repeatedly. also if you have the old one with logs grab them please.

**Comment by ct-martin** [comment]
So, what did we change between then and now?

**Comment by linux-modder** [comment]
@Tjzabel what iso was tested?  was it bare metal or virtual? so I can secondarily confirm and possibly peck at the logs and such to iron out the actual part that fixed this.

**Comment by Tjzabel** [comment]
@linux-modder The latest ISO was tested using a UEFI build box. It was installed bare-metal.

**Comment by Tjzabel** [comment]
There is no /boot/efi partition now, which is interesting

**Comment by axk4545** [comment]
@Tjzabel Make sure it wasn't done in BIOS mode by mistake

**Comment by Tjzabel** [comment]
@axk4545 How should I go about finding that out.

**Comment by axk4545** [comment]
@Tjzabel @linux-modder @ct-martin I attempted a UEFI install and was able to reproduce the issue. Please see the attached logs. Also looping @AdamWill in on this as he expressed interest in helping me debug this
[logs.zip](https://github.com/RITlug/TigerOS/files/1423938/logs.zip)

**Comment by axk4545** [comment]
This install used [linuxmodder.iso](https://tigeros.ritlug.com/isos/linuxmodder.iso)

**Comment by axk4545** [comment]
Output of `lsblk` on the effected disk:
NAME             MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sdb                8:16   0 74.5G  0 disk 
├─sdb1             8:17   0  200M  0 part 
├─sdb2             8:18   0    1G  0 part 
└─sdb3             8:19   0 73.3G  0 part 
  ├─tigeros-swap 253:3    0  7.5G  0 lvm  
  ├─tigeros-home 253:4    0 21.6G  0 lvm  
  └─tigeros-root 253:5    0 44.3G  0 lvm

**Comment by axk4545** [comment]
Will check @AdamWill on any issues the logs may reveal

**Comment by axk4545** [comment]
@Tjzabel's install was accidentally done in BIOS mode

**Comment by Josh1147582** [comment]
I installed the latest build (linuxmodder.iso) in Virtualbox, forcing EFI. `program.log` indicates that `grub2-mkconfig` is returning a nonzero status:

[](https://i.imgur.com/qCE4Llu.png)

When I looked in `/boot/efi/EFI/` after the install, there was no `tigeros` directory. I'm guessing this is the issue.

The `grub2-efi` package creates `boot/efi/EFI/fedora` automatically. If the issue is our EFI directory isn't being created, we should either use the `fedora` directory, or rename `fedora` to `tigeros`.

I'm gonna remove our EFI change, rebuild, and see if the install completes successfully.

**Comment by axk4545** [comment]
@Josh1147582 Let us know if it work and we can make the fix.

**Comment by AdamWill** [comment]
Ah, that's a reasonable theory. Other install classes that change `efi_dir`, I think, come along with a different set of bootloader packages that provide the appropriate directory. E.g. RHEL changes it to (I think) "redhat", and of course ships with grub2 packages that provide bootloader bits in `/boot/efi/EFI/redhat`.

**Comment by axk4545** [comment]
@AdamWill so do we have a need to keep it as TigerOS as far as conflicting or IP stuff? It would be nice to not repackage grub stuff

**Comment by AdamWill** [comment]
I can't speak Officially Legally, but I'd be really surprised if we considered that a branding violation. Obviously if you want to be *parallel installable* with real-Fedora, changing it is useful.

**Comment by axk4545** [comment]
@AdamWill ok. will talk it over with @RITlug/tigeros-team

**Comment by axk4545** [comment]
http://pkgs.fedoraproject.org/cgit/rpms/grub2.git/tree/grub2.spec?h=f22
So it appears that GRUB sets the efidir at build time in the specfile. Would be nice to see if this can be split out.

**Comment by axk4545** [comment]
We have implemented a temporary fix with #71 and opened #79 to track making a more permanent fix

**Comment by Rtoax** [comment]
I got 

```
The following error occurred while installing the boot loader. The system will not be bootable, Would you like to ignore this and continue with installation?

failed to write boot loader configuration
```
